File: E-bank/src/main/java/com/ebank/model/Beneficiary.java
package com.ebank.model;

import com.ebank.enums.BankName;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import jakarta.persistence.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Beneficiary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String rib;
    @Enumerated(EnumType.STRING)
    private BankName bankName;

    @ManyToOne
    @JsonIgnore
    private Account account;

    @OneToMany(cascade = CascadeType.ALL)
    private List<Transaction> transactions;
}

File: Ebank-Frontend/src/app/model/beneficiary/beneficiary.ts
import { Transaction } from '../transaction/transaction';
import { BankName } from '../../enum/bank-name';
import {Account} from "../account/account";


export class Beneficiary {
  id: number;
  name: string;
  rib: string;
  bankName: BankName;
  account: Account;
  transactions: Transaction[];

  constructor(
    id: number,
    name: string,
    rib: string,
    bankName: BankName,
    account: Account,
    transactions: Transaction[] = []
  ) {
    this.id = id;
    this.name = name;
    this.rib = rib;
    this.bankName = bankName;
    this.account = account;
    this.transactions = transactions;
  }
}
